tsinfer/algorithm.py

```diff
@@ -1118,3 +1119,166 @@ def run_traceback(self, start, end, match):
             parent[j] = e.parent
 
         return left, right, parent
+
+
+def merge_overlapping_ancestors(start, end, time):
```

Comment:
I guess we'd usually use docstrings here for implementation comments - but maybe you're consciously avoiding documenting so people don't think it's part of public API?

Reply:
Yes, that was my thought. But then I guess this won't get pulled into the actual compiled docs so even if it had a docstring it isn't actually "documented". Happy with either way.